relay-server/src/services/buffer/stack_provider/sqlite.rs

```diff
@@ -46,13 +48,23 @@ impl StackProvider for SqliteStackProvider {
         }
     }
 
-    fn create_stack(&self, project_key_pair: ProjectKeyPair) -> Self::Stack {
+    fn create_stack(
+        &self,
+        stack_creation_type: StackCreationType,
+        project_key_pair: ProjectKeyPair,
+    ) -> Self::Stack {
         SqliteEnvelopeStack::new(
             self.envelope_store.clone(),
             self.disk_batch_size,
             self.max_batches,
             project_key_pair.own_key,
             project_key_pair.sampling_key,
+            // We want to check the disk by default if we are creating the stack for the first time,
+            // since we might have some data on disk.
+            // On the other hand, if we are recreating a stack, it means that we popped it because
+            // it was empty, or we never had data on disk for that stack, so we assume by default
+            // that there is no need to check disk until some data is spooled.
+            matches!(stack_creation_type, StackCreationType::Initialization),
```

Comment:
nit: Can we move this to a helper function `assume_data_on_disk(stack_creation_type)` to clarify intent? The code comment can then be a doc comment on that function.

Reply:
Good idea, will do!